src/server/transform.rs

```diff
@@ -58,6 +58,104 @@ fn make_flux_filter_function(
+                                }),
+                                value: Some(ast::Expression::Function(Box::new(ast::FunctionExpr{
+                                    arrow: vec![],
+                                    base: ast::BaseNode::default(),
+                                    body: ast::FunctionBody::Expr(ast::Expression::Unary(Box::new(ast::UnaryExpr{
+                                        base: ast::BaseNode::default(),
+                                        argument: ast::Expression::Member(Box::new(ast::MemberExpr {
+                                            base: ast::BaseNode::default(),
+                                            lbrack: vec![],
+                                            rbrack: vec![],
+                                            object: ast::Expression::Identifier(ast::Identifier {
+                                                base: ast::BaseNode::default(),
+                                                name: "r".into(),
+                                            }),
+                                            property: ast::PropertyKey::Identifier(ast::Identifier {
+                                                base: ast::BaseNode::default(),
+                                                name,
+                                            }),
+                                        })),
+                                        operator: ast::Operator::ExistsOperator,
+                                    }))),
+                                    lparen: vec![],
+                                    rparen: vec![],
+                                    params: vec![ast::Property {
+                                        base: ast::BaseNode::default(),
+                                        key: ast::PropertyKey::Identifier(ast::Identifier {
+                                            base: ast::BaseNode::default(),
+                                            name: "r".into(),
+                                        }),
+                                        comma: vec![],
+                                        separator: vec![],
+                                        value: None,
+                                    }],
+                                }))),
+                                comma: vec![],
+                                separator: vec![],
+                            }
+                        ],
+                        lbrace: vec![],
+                        rbrace: vec![],
+                        with: None,
+                    }))],
+                    base: ast::BaseNode::default(),
+                    callee: ast::Expression::Identifier(ast::Identifier {
+                        base: ast::BaseNode::default(),
+                        name: "filter".into(),
+                    }),
+                    lparen: vec![],
+                    rparen: vec![],
+                }
+            }))
+        })
+    ));
```

Comment:
I did some manual construction of the AST when doing the test to testcase rewriter and found it rather tedious. If you intend to add more code like this do you think it could be worth to invest in a macro like https://docs.rs/quote/1.0.18/quote/macro.quote.html but for flux code?